[front] Remove ToolGeneratedFileType in favor of blob resources

philipperolet opened the pull request:
Description
---

In PR #13375 , handling of 'blob' resources, standard MCP call tool results, was introduced

This PR removes `ToolGeneratedFileType`, `isToolGeneratedFile`, `ToolGeneratedFileSchema`, in favor of those more standard 'blob' resources. Consequently, the `INTERNAL_MIME_TYPES.TOOL_OUTPUT.FILE` mimeType, that existed only for these type and schema, is also removed

It changes `CallToolResult`s of "run dust app" and "table query" (server & server v2) from `ToolGeneratedFileType` with an `INTERNAL_MIME_TYPES.TOOL_OUTPUT.FILE` mimeType, to regular blob resources. It also migrates `process.ts` to blob output.

Note: the blob resource may contain, and here does contain, a "name" and a "mimeType", used when it's then being upserted as a `FileResource`.


## Risks
Blast radius: MCP tool outputs and file generation across the platform
Risk: low

## Tests
Local tests: mcp actions process, run dust app, table query

## Deploy Plan
- Deploy front service

philipperolet committed:
Remove ToolGeneratedFileType in favor of blob resources

- Extract CSV/JSON generation logic from file creation functions
- Update MCP servers to return blob resources instead of ToolGeneratedFile
- Remove isToolGeneratedFile checks and special handling
- Update UI components to work with simplified file data
- Remove INTERNAL_MIME_TYPES.TOOL_OUTPUT.FILE constant

[front] Remove ToolGeneratedFileType in favor of blob resources

- Extracted CSV and JSON output generation from file upload logic
- Updated all MCP servers to return blob resources instead of ToolGeneratedFile
- Removed ToolGeneratedFileType, isToolGeneratedFile, and related schemas
- Simplified UI components to work with simpler file resource structure
- Removed unused INTERNAL_MIME_TYPES.TOOL_OUTPUT.FILE constant

- All MCP servers now return blob resources which are handled uniformly
- UI components updated to work with simpler props structure
- No breaking changes to external APIs

- TypeScript compilation passes
- ESLint shows no errors
- Code properly formatted

- Deploy front service

add blobcalltoolresulttype

Address PR review comments

- Rename generateJSONOutput to generateJSONSnippet (only returns snippet)
- Remove generateJSONFileAndSnippet function entirely
- Remove EXTRACT_RESULT from INTERNAL_MIME_TYPES
- Use ActionGeneratedFileType in MCPToolOutputDetails
- Add uploadToConversationDataSource flag to handleBase64Upload
- Use BlobCallToolResultBlock type in run_dust_app.ts
- Move name field outside resource object
- Add snippet field to blob resources
- Move generateSectionFile to tables_query.ts module

Address additional PR review comments

- Remove uploadToConversationDataSource flag check - always upload to conversation DS
- Replace data URI substring with 'transient-content' placeholder
- Use BlobCallToolResultBlock type in tables_query server_v2
- Fix blob resource structure in server_v2 with name field outside resource

fix extract details

update blob uri

Fraggle commented on front/lib/actions/action_file_helpers.ts:
```diff
@@ -6,7 +6,8 @@ import { processAndUpsertToDataSource } from "@app/lib/api/files/upsert";
 import logger from "@app/logger/logger";
-import type { CoreAPIDataSourceDocumentSection } from "@app/types";
+
+export { generateCSVSnippet } from "@app/lib/api/csv";
```
isn't it weird to import-export ?

Fraggle commented on front/lib/actions/mcp.ts:
```diff
@@ -873,16 +874,34 @@ export class MCPConfigurationServerRunner extends BaseActionConfigurationServerR
+        const file = uploadResult.value;
+
+        // Upload the file to the conversation data source if requested
+        if (uploadToConversationDataSource) {
```
why is it an option ?

Fraggle commented on front/lib/actions/mcp.ts:
```diff
@@ -873,16 +874,34 @@ export class MCPConfigurationServerRunner extends BaseActionConfigurationServerR
+                    blob: "",
+                    // For blob resources, URI can be updated now that the file
+                    // is uploaded.
+                    uri: file.getPublicUrl(auth),
```
I'm not sure this url is persistent.

Fraggle reviewed (approved):
Looks fine but it's hard to tell from all the changes.
Trusting you and keep an eye on the monitors :)

philipperolet commented:
Thanks a lot for the review!

Kindly asking for confirmation: When we had discussed blob resource, you explained how removing TOOL_OUTPUT.FILE was a valuable refactor that we should do if not directly. I agreed, but we only talked about it for ~3mns so I'm not entirely sure that the PR is what you had in mind. This is why I would appreciate a confirmation from you that we agree this is more or less what needed to be done?

In general, if you have hints on how I could have made the PR more easily reviewable so that you could have felt more confident in your review, I'd gladly take them :pray:

Fraggle commented:
> Kindly asking for confirmation: When we had discussed blob resource, you explained how removing TOOL_OUTPUT.FILE was a valuable refactor that we should do if not directly. I agreed, but we only talked about it for ~3mns so I'm not entirely sure that the PR is what you had in mind. This is why I would appreciate a confirmation from you that we agree this is more or less what needed to be done?

This PR is what I had in mind indeed.

> In general, if you have hints on how I could have made the PR more easily reviewable so that you could have felt more confident in your review, I'd gladly take them 🙏

I'm not blaming you :)
You switch 3 tools (process, dust app and table query) all at once + refactored a bit of code along the way so it's hard to be sure that everything will be working.

Things that I had in mind:
- What about the previous output, how will it looks ? (I don't mind if it's degraded as long as it's not broken).
- (at some point, was it still the case ?) There was the "snippet" thing with the same name for 2 differents purposes : in table query, it was presented to the model but in the upsert to datasource part, it was my "hint" that the file was upserted. Did we keep the 2 purposes working ?

philipperolet commented:
Very clear thanks, will check out specifically the points you mentioned (previous output, snippets) in testing

philipperolet committed:
Fix TypeScript errors and remove ExtractResultResourceSchema

philipperolet committed:
Fix blob resource snippet not being set on FileResource

The snippet from blob resources was not being transferred to the FileResource object, causing the preview to show incorrect content. Now the snippet is properly set on the file resource when processing blob uploads.
